egui/src/ui.rs

```diff
@@ -251,6 +257,12 @@ impl Ui {
         &self.layout
     }
 
+    /// Mutably borrow internal `Layout`.
+    /// Changes apply to this `Ui` and its subsequent children.
+    pub fn layout_mut(&mut self) -> &mut Layout {
```

Comment:
The design of how `Layout` works is not yet complete. In particular, when changing `ui.layout` one must also take care that `ui.cursor` is still valid for that new layout. The `Ui::set_layout` tries to do this, but I'm not quite happy with the design yet (which is why `Ui::set_layout` is marked with `// TODO: remove`). I think `cursor` may be removed from `Ui` and moved to some Layout-specific layout state.

Long story short: if you want to change layout you should use helpers like `ui.horizontal(...)` or `ui.set_layout`, but accessing the `Layout` directly via a `layout_mut()` is not a good idea.